<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Peter Hraber" />

<meta name="date" content="2018-10-24" />

<title>Swarm Selection Tutorial</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">Swarm Selection Tutorial</h1>
<h4 class="author"><em>Peter Hraber</em></h4>
<h4 class="date"><em>2018-10-24</em></h4>



<div id="phase-i-select-sites" class="section level1">
<h1>Phase I: Select Sites</h1>
<p>This phase of analysis requires:</p>
<ol style="list-style-type: decimal">
<li>A protein alignment, provisionally assumed to be in FASTA format.</li>
<li>A way to recognize how the longitudinal samples are labeled.<br />
By default, sequence names are assumed to be dot-delimited, with the timepoint label in the first (left-most) field.</li>
<li>An indication of which sequence is the reference/TF sequence.<br />
By default, this is taken to be the first sequence in the alignment.</li>
</ol>
<div id="choose-a-cutoff-setting" class="section level3">
<h3>Choose a Cutoff Setting</h3>
<p>We need to choose a setting for the cutoff parameter value. Let’s see how number of sites depends on cutoff threshold. A vector of cutoff values shows how many sites result from multiple settings. Note that the default values work for these data.</p>
<p>In practice, you will not need to use <code>system.file()</code> unless you are referring to an example alignment included with the <code>swarmtools</code> package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">    <span class="kw">library</span>(lassie)
    alignment_file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;CH505-gp160.fasta&quot;</span>, <span class="dt">package=</span><span class="st">&quot;lassie&quot;</span>)
    eg.swarmtools &lt;-<span class="st"> </span><span class="kw">swarmtools</span>(<span class="dt">aas_file=</span>alignment_file, <span class="dt">tf_loss_cutoff=</span><span class="dv">80</span>)
    <span class="kw">summary</span>(eg.swarmtools)</code></pre></div>
<pre><code>##   %TF Loss at least selected sites, n selected sites, %
## 1                80                35              3.67</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">    <span class="kw">plot</span>(eg.swarmtools)
    <span class="kw">print</span>(eg.swarmtools)</code></pre></div>
<pre><code>## Loss cutoff = 80%.
## Selected 35 sites:
##       aln   l   r aa peak_tf_loss when_up   tf_area
## N279  357 279 279  D        95.83       4  3423.514
## H417  499 417 417  P        91.18       7 10118.671
## V281  359 281 281  A       100.00       9  1732.931
## T413  495 413 413  T        88.24       9  6179.279
## N332  412 332 332  O       100.00      20  1734.220
## O334  414 334 334  S       100.00      20  1734.220
## -144g 214 144 145  -       100.00      20  2093.700
## -144h 215 144 145  -       100.00      20  2093.700
## -144f 213 144 145  -       100.00      20  3408.520
## V756  844 756 756  I        92.86      20  4903.194
## A145  216 145 145  G        96.77      20  8750.560
## D325  405 325 325  N        83.33      20 10922.040
## Y330  410 330 330  H       100.00      22  2806.340
## N300  379 300 300  N       100.00      30  2868.218
## T234  312 234 234  O       100.00      30  2917.280
## K302  381 302 302  N       100.00      30  2917.280
## -465  552 465 465  S       100.00      30  3071.900
## -464  551 464 464  E       100.00      30  3145.650
## N462  544 462 462  N        89.29      30 11450.590
## -463e 550 463 464  -       100.00      53  5433.226
## K460  542 460 460  N       100.00      53  6289.694
## O398  480 398 398  S        91.30      53  7020.274
## K347  427 347 347  S        83.33      53  8187.620
## I151  222 151 151  K        83.33      53  8298.458
## H356  437 356 356  O       100.00      78  6617.376
## E275  353 275 275  V        91.67      78  8185.790
## G471  558 471 471  G        87.50      78  8966.026
## O130  193 130 130  K        87.50      78  9129.296
## O147  218 147 147  M        91.67      78  9312.777
## E640  727 640 640  S        83.87      78  9441.390
## T132  195 132 132  T        83.33      78  9724.856
## D185  259 185 185  D        83.33      78 10942.110
## M4      4   4   4  K        87.50     100 10805.110
## G620  707 620 620  E        91.67     136 10612.342
## R412  494 412 412  D        83.33     160 13949.102</code></pre>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAC+lBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///9JINbqAAAACXBIWXMAAA7DAAAOwwHHb6hkAAATHUlEQVR4nO2dCVhV1dqAF4MiMk+ioKKI4nCV4ngVRVNTcyrMkcoKzByut7+bGFkqXUkvmulfjqQ55PXibKmp99e0ciAFFTNFRSBHVBBSUUHgfM/zrz2cSc7hW+ewBWqv93k6ezjrfH68nbP3PvustT4CnCohtZ1AXYcLQuCCELggBC4IgQtC4IIQuCAELgiBC0LgghC4IAQuCIELQuCCELggBC4IgQtC4IIQuCAELgiBC0LgghC4IAQuCIELQuCCELggBC4IgQtC4IIQ6oqgEYSU2fCyc8Oae/ddL6xVOBORFXT9SP/G7p2X2xKvMnVKUFpCQpZVr9rfQJSygK5eJXpBO6S1sYokVlcETQkLK4NlhOyx5kUlrYnnolUhxOMewAHSMkHgBGibkcYr1wcTclKJxOqKIBFrBR0W3zBnCPlfgGTyhryXvpeWA2wk5EslcqpJQRoSWBjT3HdYrn7P3r6NXf7yQQFdG0CcoK/wwegKcGF0sMdzS7SmDcyxkpDzdOEnuIkjM+W9p3v1ou+d3YTsUCLpmhXkrREctCmVd6yTDhbt7kmC3vAlpPEI+N5F3Dtca9LAHMskQY1Ib4Ch5O1Il3ZxD+Wn8vuT5neVSLpmBRH/hDmNCNkp7wginmt3jSNkviRI+oiVtSfNd6a/Scg2kwbm2E/IMoAMQloBdJBc/lU6d/W0I95XFEm6hgX9AvA1IZ9J28X0A/U7aMfHJBsL+i8hGwDKAslzJg1EQuVTFSkSN+/7EZc5i5sTEgIVDUjj1d90FQ8/IOka8kiJpGtWkB99PEZIorRd4UdIgyFLfxPWDYK+IKRpq1atGpBGJg1EnhAEXwnrju4kEkrWrfsVIK8+GSw+8fBMNCHTlUi6hg/S9DFNLwgOhoh/bd8rxoLidBLIfeMG5tne3bXTnhAyXLfdgQTLa4+9SEclkq5VQVDx47tBVMDzxoLmiR/ESg0s88iZvA+XDx4UjuRtSCf4Niwsg662ox88BahNQamTJ58B7ZFA4mIQtBu2E7KSPnn60M8mDUSe+IhlRUZ+KZ7qTkEKISkAmfbkVeHqKAkg25FEKZF0bQo6QU86639Y4UXCZEErCJld9iCQBGz7eWF9+lXBuIHIE4JKXYn7e1OdSUctXK5PvGf+swkhB+BBU1Lvf2bTQ/dqJZKuTUHaV+U/99+yoJ/EC8Wt9cSdba+bNDDLGvHZ+ofo6mdSy3fp6g5HcXWIVomka/UYVLoyskm9pgOFrxeiIO0HvsKV9MmXmjVs/3GhaQPz/F8vT88hJ8TVvX18/fpuE1dPvhjo0X1JhSJJ16nvYnURLgiBC0LgghC4IAQuCIELQuCCELggBC4IgQtC4IIQuCAELgiBC0LgghC4IAQuCIELQuCCELggBC4IgQtC4IIQuCAELgiBC0LgghC4IAQuCMEKQRti6MO1ge4dd+oXKoBR0Km0EzMaCYK6jLu5wSlHt1ABbII+iYh8aXIYFXS6XhHAoI/lxVPOrU6ACioXHgLv3PWBKVTQ5lC69eFr8uJpJ1cXQAX1+1YL0GNtikYUlNyF7prXX16YttwbHx/fJ2bS08mz1kAF3U8ceBByRg+/IAra0pbu+ugVeWHa8nhSUlJgVIenk2etwXAMKpg6NE1YCoJ+cboH8NJMeVG5bURypNIZ1jL4MWjr1vJrE6PPSoKgy9TS/Q1zdItKqFDQhBEj/waQNWasJOjaAK8Ou/SLSqhQUOO7dwOEZQZTOBUKGpw460X2cCoUlB831dKQUTOoUJB1cEEIXBCCegVd25lWztBMvYI2u+xayNBMhYLOwFkrwqlQkNepgCwRpnAqFJTU1jFUhCmcCgXdgiArwqlQkGdes/siTOFUKCjO085DhCmcCgUBvGxFOFUKAijPY50wTZWCbo5ybOg44hZTOFUKGjr2NuTHsn3QVCnITZgD4Y67uL7KSaARTBIWGyu3VaWg1vvpw37pQrEokxITB73n06WZOcJUKWizR2xCrMcW/fYvvR9D4DHzbVUpqDh7YfzCS8W6zfJuZ6CYvNaqyyIzs4eoUFBZWfMySqGbbsfCtwEymy3O2uaz2LTlYo1G4zpXdYIcHIiDgO535vKATHntk56mLfPT09M7LledIADTPgr7e9CHk5vpw4K+lduq8CP2JLFz6cNpxzVFx4O+qvysKgV1lZA2Hrr9LCw2tnVqvcjMFFiqFJSamnp0U69NTOFUKUjkThjeBtQsKMMNbwMqFSQcgDSO/2AKp0pBqQKZbJMSqlJQ6ZJ9sP3TEqZwqhT0ZngaZPRgqzGgSkGu2fThMr9pb5FWwqVhajDaTkCVglJ83pn/D981TOFUKQjOz5owk60Pp0oFWQEXhMAFIahUUPkN1sm9VSnoRj9n19yu2UzhVCno1YkP/CtmVlm5QY8qBXkXgT8UNmQKp0pBnXZSQQfkgXLyb86WRj2rUtAPPiOdY/3kygTyb86WRj2rUhAUrElccUNel35ztjjqWZWCor8x1KGSf3O2OOpZlYLmdfeM3ftYWpd/czY76nkWIcQ+SYWCAPKSBwZMMGx+0tPsqGcBVb6DAB7t+Xtge3FN/s2Zj3o2JjnKNWLuBWld95szH/VsxIBlNwwb8m/OfNSzCdZ2A17dwvXlfGEz08V4S2KPE4CDMnWYa4Kn0Q34UoPjv/X9Gwid0RyMtmSur/+zCbK6G/DaHgAbhG5EC0c5GG1B+SRPn1mQGQr9SfPiH8O8X7sh76IMSKbXDdGwNLBBxEXI7JXYcew8gJnv2P6HKYWV3YAxREEV5RXnx06l753QbAfDFsDmNjknnS5RQfQdVOCzo3BCP3kXfW5ZFED37bfq/5AfMwEyPSae3tqbfgv8sRp/mUJY2Q0YQz5IbyBNbkFFn135DvotSkrIacgvlQStGQnwwLlC2kWfu+H6KM/z0cMcKJkWDZmupXDP5fdsf5Zhsk8ZG7oBV4XuLFbwrzBIHgOSIHGLUrYgqOX8EklQorO/v7/rDWmX+Mo9yTFQMSci8nkqSCgA+sLmz+rCXEQsZzGxGzBbOFHQiiUAhaQk2s3Hi/ikylv0yZzL2lPPpEiCVsTSD1+uVtolvHLu3wftgY2aW/B1tHCYAvgituf3Nv9ZysEiqDzzh2zGm9KioO0BF67HaaDg6tXT9ldL5C3KvLBrv2lWiIIKr/sdLEroKu8SnrwQGPAYPu92+6imf7koKMfLvy6c6xgEZYT6hnt1Ybsn3fX15qdB+5G/6wsXhU3hI2bYuvuSi9c48Rj0ilvx7nbOfS7Ju8SXtqff9gqfd47YHbROFAQd3rbtT1IWBkGdZ5RBaVwvpnBBH0dOv1nNlHR0+69CgaoFy2leGK7KeJpv+rk9IW2rl5JE8cGAx0rEqS4MgkavpQ8rhzGFa/Hlg6lf2Nkvql5SAtv8t1U/iAKggsaMGU40I8LtxzOFi5g9lx6sRm6udl51BlTQOh1M4SJShUdVCZIpt0ZQhz/RZIoMgi5OiI6O7t+CKZwkaFV4tXKqU7D0k4754Pm14buZwkmCDrF1y/9DwCDIqaAkEg53YwonCfrJge2y8o8Ag6CAn6Dr7Vvyb/PHIl2aTXtsedSzKAj8zimYYu3CIGhJ/ZzZmm7S6LniJtPvngxOtDzqWRLk8bmSOdYqLGexWw/LN3xRJK4ecS2nX7x7WR71LAmK+lah9GofK7vgFV8FqBg8zeyo56v79u1rf0RcDbNi8s46jvV9FHNe7HbT7Kjnr/r16+ch9YmZ6KfUN9Zax1pBJTP9EuRxLU+OehaQP2JnfNiuCv4AsAjKKi1JWSHdtNFG9c+DqkY9S4Ig8rBSCdY2DIL+6ZQ/7xmNdPdqv9vZ3NzcPMujnmVBbRYomGOtwtJH8ZS2ZUaut7j+CRHoZXnUsyxI86qCOdYqDII8cjOCtFddmcLpBM2aUY2c6hQMgia2CVlwOSyKKZwqBZVtTinLmWfmstkMOkGxTRJsTql63CtSNp7SQxFkQWcHd0q1MaPqcS/yLWUDKj0UQaflPyH/sj0p27kX+ewdZSMqPRRB/77pbN/4fZuzsplhwXe03ySdVDCi0kMRDB+s1U3tnb322JqXjRxscGiCZ097BUufWTsUASHC+Mjz61Lnhi5v/rumDkZlGxcVwnzvxvkwrY9yUa0dioAQYWrjzqHQZr7EYa8tmVlL+YCQ0O6gHTGWfm1+TrmwLJO8iUMRivGGUEmQRKBdzNz0tAvWJWY1O5o+zG+Ycvne/QWLff+jXFjrJ3mrErOCtJ39G2hCHHybfbr3Kb6XNgbeH2PXpP73OSP7KujH+kneqsasIIktw91aP+PUqt5br4QnJY0bN2lJdnb2r4WUUvZsq+LB+AcbsyveUvDTJWL1JG9VU4UggYpDB6KS5/SOjx/Q07d+cHCAg5eXl6cdffBy6KTRtPMclEyZwXbRbp7krtV4sTmUHvVswymrQHgbHU5PT98f1Xk8xcXO3cvLveU+yuYHVkXaFL3Id731CVQJKshpfag1k/4rcU4vosJ2BvejNGypCemQOJ31haeHRbL9Qm4FqKA91zMlmMIpIsjAqePpyVFv2JH6LWZlmZkYtSbAP2Ja+vX41LJf2MIpLEimbE9kYydi1zS4uZMmuOmoLnFmW2XFffyp8v82KuhC6MtwrkEf9++Ywj0dQRJ59LS3Nf27WZvCiQl2vk7vx8fH/3XgsuQJHyYv/0jZfxUVNCS+AqI/h6XdTfZaHPVcCzc5bmS9nkT5yzPjxw6OGh/r0GgUpfNb6w4UFlpROs4SeF2Nm1Du+zucM+2jaHHUc+3cBRJJnTqQ/s+Er6duogwKalTf3Ys0HB9FLxwmrV5MT5K2Xcujgtyz4Gg3gIvexjstj3quRUFQVun+yq+jJofRC4dWbVw1Gk3bQFuCooIGTCkeOQdg5gvGO82Oej5E3+fN99uShFIkVPnsvqAkI1i7iKKCzrYkHe8/6u2TbrzT7KjnA/RIOfoK47/7VEio8tkro+KNYL1gwk/zFXlaKN1y22SfxVHP9IyczsJPB5ma7TnG1Ow7ebmjylbHjzM6sVKQOSyOeoZv3DQsBDZhaubRhqmZfThLq7ZNa06QxVHPcJhtUO/sD5mavcA2GsGJaZrQo2y9CJ/ARkGWRj1zQRhcEMLN5UzNDrONlVt1lalZItPvmreXMQV7AqUF/engghC4IAQuCEFhQZZugxiB9NXXgcwmJ4PVO5PYEGNIjSFDExQWZOk2iAGsr74OZDY5GazemcCJGY1iDKnhGZqirCCLt0EMYH31dSCzyRlRVb0zgTWTw2L0kVgCmqCsIIuTvxmoqq++cTNkNjkDVdY7k5gSo0+NIaApygoyexukEhb76htR1Wxyplisd2ZAECRHYsvQCGUFWb4NYgDpq29MFbPJ6bFc78yAIEiOxJKhCcoKsnwbRA/WV18Gm01OT9X1ziQEQXIkhgxNUfosZuk2iB6sr74MNpucnqrrnUlMiTGkhmdoitLXQZZug+jB+urrQGaT04HUO5MQBcmR8AxN4VfSCFwQAheEwAUhcEEIXBACF4TABSFwQQhcEAIXhMAFIXBBCFwQAheEwAUhcEEIXBACF4TABSFwQQg1IOg+yccbSRgKksR53kryPiOuFrzYZFI53O9nWiKBNsCinVJiQtC6KsgzD9qeklanJt4dsQ0Sn5h50DMPjfZHE7SljfuwW3ItFn1FFtgQ4hb7OLUrAP1PqNgitXrZrlmQY8BZsUVc4r2RWwsG6H710je4rav1Itdzyeo+xSfyaGfX9yBl3Bse3TJFQVIBmGpQc4KyPfbdiR0t12LRV2Q575t66dmlsiD6DpJbgcd9CM2SXp8/hH7Eph6Qoxk10NV6keu5ZNl9dUfjf/kIuZPiuOh2fCctFSQXgKkGNSdoYQzAbcdyqRaLviJL4jsAafsMguRWxoIErg3VrRk10NV6keu5ZHmXQfxkgMDclA5aeOx9kQqSC8BUI/uaExSfQB8a5km1WPQVWSbMF54T5BwVBcmtnhQ08XjOc01nG4WRBElh5HouWaEAHyUCBOWmCDrDD1JBcgGYamRfg++gWHpCciyTarHoK7LMfA8gY3uqBmCT9A6SWj0hKCsaRq8tiMgwhJEESWHkei5GgsKpO7/zVJBcAKYa2deIoEtFRUUllzy+L4wdLtdi0VdkOeV77EqPTzMdTxf2EgQVyq1kQZuvSSFePwvDN9yLTKOrRg10tV7kei5GgsiaoukdK6gguQBMNagRQQIrYFNrt6E35Voshoosq1q4x5Zo33XttLGrWLFFaiULcpH6o2aMpUfzZ72niRuGBrpaL3I9FyNBg4a7RpwTz2JSAZhqULevpHel2fSylGjlUqjbghbYVmBMPYJshAuqQbggBC4IgQtC4IIQuCAELgiBC0LgghC4IAQuCIELQuCCELggBC4IgQtC4IIQuCAELgiBC0LgghC4IAQuCIELQuCCELggBC4IgQtC4IIQuCAELgiBC0LgghC4IAQuCIELQuCCELggBC4IgQtC4IIQuCAELgiBC0LgghC4IIT/B0ghpnifqJV6AAAAAElFTkSuQmCC" /><!-- --></p>
<p>The 80% cutoff value gives 35 sites. Let’s go with that.</p>
</div>
</div>
<div id="phase-ii-select-clones" class="section level1">
<h1>Phase II: Select Clones</h1>
<p>This phase of analysis requires a SwarmTools object created in Phase I.<br />
The SwarmTools object must have a list of selected sites, which happens only when it was created using a single <code>tf_loss_cutoff</code> value. Let’s just go with the defaults.</p>
<pre><code>eg.swarmset &lt;- swarmset(eg.swarmtools)</code></pre>
</div>
<div id="all-together-now" class="section level1">
<h1>All Together Now</h1>
<p>Got that? Here’s the whole workflow:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">    <span class="kw">library</span>(lassie)
    alignment_file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;CH505-gp160.fasta&quot;</span>, <span class="dt">package=</span><span class="st">&quot;lassie&quot;</span>)
    eg.swarmtools &lt;-<span class="st"> </span><span class="kw">swarmtools</span>(<span class="dt">aas_file=</span>alignment_file, <span class="dt">tf_loss_cutoff=</span><span class="dv">80</span>)
    <span class="kw">summary</span>(eg.swarmtools)</code></pre></div>
<pre><code>##   %TF Loss at least selected sites, n selected sites, %
## 1                80                35              3.67</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">    <span class="kw">print</span>(eg.swarmtools)</code></pre></div>
<pre><code>## Loss cutoff = 80%.
## Selected 35 sites:
##       aln   l   r aa peak_tf_loss when_up   tf_area
## N279  357 279 279  D        95.83       4  3423.514
## H417  499 417 417  P        91.18       7 10118.671
## V281  359 281 281  A       100.00       9  1732.931
## T413  495 413 413  T        88.24       9  6179.279
## N332  412 332 332  O       100.00      20  1734.220
## O334  414 334 334  S       100.00      20  1734.220
## -144g 214 144 145  -       100.00      20  2093.700
## -144h 215 144 145  -       100.00      20  2093.700
## -144f 213 144 145  -       100.00      20  3408.520
## V756  844 756 756  I        92.86      20  4903.194
## A145  216 145 145  G        96.77      20  8750.560
## D325  405 325 325  N        83.33      20 10922.040
## Y330  410 330 330  H       100.00      22  2806.340
## N300  379 300 300  N       100.00      30  2868.218
## T234  312 234 234  O       100.00      30  2917.280
## K302  381 302 302  N       100.00      30  2917.280
## -465  552 465 465  S       100.00      30  3071.900
## -464  551 464 464  E       100.00      30  3145.650
## N462  544 462 462  N        89.29      30 11450.590
## -463e 550 463 464  -       100.00      53  5433.226
## K460  542 460 460  N       100.00      53  6289.694
## O398  480 398 398  S        91.30      53  7020.274
## K347  427 347 347  S        83.33      53  8187.620
## I151  222 151 151  K        83.33      53  8298.458
## H356  437 356 356  O       100.00      78  6617.376
## E275  353 275 275  V        91.67      78  8185.790
## G471  558 471 471  G        87.50      78  8966.026
## O130  193 130 130  K        87.50      78  9129.296
## O147  218 147 147  M        91.67      78  9312.777
## E640  727 640 640  S        83.87      78  9441.390
## T132  195 132 132  T        83.33      78  9724.856
## D185  259 185 185  D        83.33      78 10942.110
## M4      4   4   4  K        87.50     100 10805.110
## G620  707 620 620  E        91.67     136 10612.342
## R412  494 412 412  D        83.33     160 13949.102</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">    eg.swarmset &lt;-<span class="st"> </span><span class="kw">swarmset</span>(eg.swarmtools)</code></pre></div>
<pre><code>##     TF w000.TF        NHVTNO---VADYNTK--N-KOKIHEGOOETDMGR</code></pre>
<pre><code>## Number of mutations to be represented is now 92</code></pre>
<pre><code>## t=w004, n=8 viable clones</code></pre>
<pre><code>##   Checking for mutations in column  1: N357[KD]</code></pre>
<pre><code>##     B1 w004.54.KC247604   K..................................</code></pre>
<pre><code>## Number of mutations to be represented is now 91</code></pre>
<pre><code>##   Checking for mutations in column  3: V359[SGDA]</code></pre>
<pre><code>##   Checking for mutations in column 22: O480[T-]</code></pre>
<pre><code>##     B1 w004.31.KC247583   .....................-.............</code></pre>
<pre><code>## Number of mutations to be represented is now 90</code></pre>
<pre><code>## t=w007, n=16 viable clones</code></pre>
<pre><code>##   Checking for mutations in column  1: N357[D]</code></pre>
<pre><code>##   Checking for mutations in column  2: H499[SRN]</code></pre>
<pre><code>##     B1 w007.8.KM284749    KR.................................</code></pre>
<pre><code>## Number of mutations to be represented is now 89</code></pre>
<pre><code>##   Checking for mutations in column  4: T495[OI]</code></pre>
<pre><code>##     A0 w007.34.KM284744   ...I...............................</code></pre>
<pre><code>## Number of mutations to be represented is now 88</code></pre>
<pre><code>##   Checking for mutations in column 29: O218[SNKD]</code></pre>
<pre><code>##     A0 w007.25.KM284734   ............................N......</code></pre>
<pre><code>## Number of mutations to be represented is now 87</code></pre>
<pre><code>##   Checking for mutations in column 35: R494[Q-]</code></pre>
<pre><code>##     A0 w007.21.KM284732   ..................................Q</code></pre>
<pre><code>## Number of mutations to be represented is now 86</code></pre>
<pre><code>## t=w008, n=12 viable clones</code></pre>
<pre><code>##   Checking for mutations in column  1: N357[D]</code></pre>
<pre><code>##   Checking for mutations in column  2: H499[SN]</code></pre>
<pre><code>##   Checking for mutations in column  3: V359[SGDA]</code></pre>
<pre><code>##     A0 w008.20.KM284762   ..A................................</code></pre>
<pre><code>## Number of mutations to be represented is now 85</code></pre>
<pre><code>##   Checking for mutations in column 29: O218[SKD]</code></pre>
<pre><code>##   Checking for mutations in column 35: R494[-]</code></pre>
<pre><code>## t=w009, n=14 viable clones</code></pre>
<pre><code>##   Checking for mutations in column  1: N357[D]</code></pre>
<pre><code>##   Checking for mutations in column  2: H499[SN]</code></pre>
<pre><code>##   Checking for mutations in column  3: V359[SGD]</code></pre>
<pre><code>##     A1 w009.19.KM284781   ..G................................</code></pre>
<pre><code>## Number of mutations to be represented is now 84</code></pre>
<pre><code>##   Checking for mutations in column  4: T495[O]</code></pre>
<pre><code>##   Checking for mutations in column 35: R494[-]</code></pre>
<pre><code>## t=w010, n=13 viable clones</code></pre>
<pre><code>##   Checking for mutations in column  1: N357[D]</code></pre>
<pre><code>##   Checking for mutations in column  2: H499[SN]</code></pre>
<pre><code>##     A1 w010.7.KM284714    .N.................................</code></pre>
<pre><code>## Number of mutations to be represented is now 83</code></pre>
<pre><code>##   Checking for mutations in column  3: V359[SD]</code></pre>
<pre><code>##   Checking for mutations in column  4: T495[O]</code></pre>
<pre><code>## t=w014, n=6 viable clones</code></pre>
<pre><code>##   Checking for mutations in column  2: H499[S]</code></pre>
<pre><code>##   Checking for mutations in column  3: V359[SD]</code></pre>
<pre><code>##   Checking for mutations in column  4: T495[O]</code></pre>
<pre><code>## t=w020, n=23 viable clones</code></pre>
<pre><code>##   Checking for mutations in column  2: H499[S]</code></pre>
<pre><code>##   Checking for mutations in column  3: V359[SD]</code></pre>
<pre><code>##   Checking for mutations in column  4: T495[O]</code></pre>
<pre><code>##   Checking for mutations in column  5: N412[O]</code></pre>
<pre><code>##     B0 w020.15.KC247489   ....OS....T........................</code></pre>
<pre><code>## Number of mutations to be represented is now 80</code></pre>
<pre><code>##   Checking for mutations in column  7: -214[TOIA]</code></pre>
<pre><code>##     B1 w020.25.KC247496   .R....ATO..........................</code></pre>
<pre><code>## Number of mutations to be represented is now 77</code></pre>
<pre><code>##   Checking for mutations in column  8: -215[VSKEA]</code></pre>
<pre><code>##   Checking for mutations in column  9: -213[IA]</code></pre>
<pre><code>##   Checking for mutations in column 10: V844[A]</code></pre>
<pre><code>##     B0 w020.24.KC247495   .RA......AT........................</code></pre>
<pre><code>## Number of mutations to be represented is now 76</code></pre>
<pre><code>##   Checking for mutations in column 11: A216[VSGD-]</code></pre>
<pre><code>##   Checking for mutations in column 12: D405[N]</code></pre>
<pre><code>##     C1 w020.11.KC247485   ..........TN.......................</code></pre>
<pre><code>## Number of mutations to be represented is now 75</code></pre>
<pre><code>##   Checking for mutations in column 29: O218[SKD]</code></pre>
<pre><code>## t=w022, n=15 viable clones</code></pre>
<pre><code>##   Checking for mutations in column  2: H499[S]</code></pre>
<pre><code>##   Checking for mutations in column  3: V359[SD]</code></pre>
<pre><code>##   Checking for mutations in column  4: T495[O]</code></pre>
<pre><code>##     A0 w022.22.KM284717   ...O...............................</code></pre>
<pre><code>## Number of mutations to be represented is now 74</code></pre>
<pre><code>##   Checking for mutations in column  7: -214[TOI]</code></pre>
<pre><code>##   Checking for mutations in column  8: -215[VSKEA]</code></pre>
<pre><code>##   Checking for mutations in column  9: -213[IA]</code></pre>
<pre><code>##   Checking for mutations in column 13: Y410[H]</code></pre>
<pre><code>##     B1 w022.6.KC247523    ..AIOSATO...H......................</code></pre>
<pre><code>## Number of mutations to be represented is now 73</code></pre>
<pre><code>##   Checking for mutations in column 14: N379[SG]</code></pre>
<pre><code>##     A0 w022.5.KC247522    .RA.OSATO....S.....................</code></pre>
<pre><code>## Number of mutations to be represented is now 72</code></pre>
<pre><code>##   Checking for mutations in column 24: I222[VL-]</code></pre>
<pre><code>##     A0 w022.9.KC247525    ..GIOS.................-...........</code></pre>
<pre><code>## Number of mutations to be represented is now 71</code></pre>
<pre><code>## t=w030, n=24 viable clones</code></pre>
<pre><code>##   Checking for mutations in column  1: N357[D]</code></pre>
<pre><code>##     A0 w030.17.KC247532   D.GOOSATO.......TD.................</code></pre>
<pre><code>## Number of mutations to be represented is now 68</code></pre>
<pre><code>##   Checking for mutations in column  2: H499[S]</code></pre>
<pre><code>##   Checking for mutations in column  3: V359[SD]</code></pre>
<pre><code>##     A1 w030.13.KC247529   ..DPOS.............................</code></pre>
<pre><code>## Number of mutations to be represented is now 67</code></pre>
<pre><code>##   Checking for mutations in column  7: -214[TOI]</code></pre>
<pre><code>##   Checking for mutations in column  8: -215[VSKEA]</code></pre>
<pre><code>##   Checking for mutations in column  9: -213[IA]</code></pre>
<pre><code>##   Checking for mutations in column 11: A216[VSGD-]</code></pre>
<pre><code>##     A0 w030.26.KC247539   .RG.OS....-....NTD.................</code></pre>
<pre><code>## Number of mutations to be represented is now 65</code></pre>
<pre><code>##   Checking for mutations in column 14: N379[G]</code></pre>
<pre><code>##   Checking for mutations in column 15: T312[O]</code></pre>
<pre><code>##     A1 w030.36.KC247549   ....OSATO.....O.TD.................</code></pre>
<pre><code>## Number of mutations to be represented is now 64</code></pre>
<pre><code>##   Checking for mutations in column 17: -552[PF]</code></pre>
<pre><code>##   Checking for mutations in column 18: -551[TNGE]</code></pre>
<pre><code>##     A0 w030.20.KC247541   ..AIOSATOA......TNTO...............</code></pre>
<pre><code>## Number of mutations to be represented is now 61</code></pre>
<pre><code>##   Checking for mutations in column 19: N544[SOD-]</code></pre>
<pre><code>##     A1 w030.32.KC247546   .RG.OSATO.......TD-................</code></pre>
<pre><code>## Number of mutations to be represented is now 60</code></pre>
<pre><code>##   Checking for mutations in column 20: -550[TKE]</code></pre>
<pre><code>##   Checking for mutations in column 21: K542[ONE-]</code></pre>
<pre><code>##     B0 w030.21.KC247535   .RA.OSATO...........E..............</code></pre>
<pre><code>## Number of mutations to be represented is now 59</code></pre>
<pre><code>## t=w053, n=21 viable clones</code></pre>
<pre><code>##   Checking for mutations in column  2: H499[S]</code></pre>
<pre><code>##   Checking for mutations in column  3: V359[S]</code></pre>
<pre><code>##   Checking for mutations in column  7: -214[TOI]</code></pre>
<pre><code>##     B1 w053.22.KC247620   DRGIOSIEIAG.HSONFT.E.TE............</code></pre>
<pre><code>## Number of mutations to be represented is now 50</code></pre>
<pre><code>##   Checking for mutations in column  8: -215[VSKA]</code></pre>
<pre><code>##   Checking for mutations in column  9: -213[A]</code></pre>
<pre><code>##   Checking for mutations in column 11: A216[VSD]</code></pre>
<pre><code>##   Checking for mutations in column 14: N379[G]</code></pre>
<pre><code>##   Checking for mutations in column 17: -552[P]</code></pre>
<pre><code>##   Checking for mutations in column 18: -551[GE]</code></pre>
<pre><code>##   Checking for mutations in column 20: -550[TK]</code></pre>
<pre><code>##   Checking for mutations in column 21: K542[ON-]</code></pre>
<pre><code>##     A1 w053.31.KC247628   DRGIOSAT....HSON....N-.............</code></pre>
<pre><code>## Number of mutations to be represented is now 49</code></pre>
<pre><code>##   Checking for mutations in column 24: I222[VL]</code></pre>
<pre><code>##     B1 w053.15.KC247614   D.G.OSATOA..HSONFT.E.-.L...........</code></pre>
<pre><code>## Number of mutations to be represented is now 48</code></pre>
<pre><code>##   Checking for mutations in column 25: H437[QONGD]</code></pre>
<pre><code>##     A0 w053.8.KC247632    DRGIOSATOA..HSONFT.E.T..Q..........</code></pre>
<pre><code>## Number of mutations to be represented is now 47</code></pre>
<pre><code>##   Checking for mutations in column 27: G558[VE]</code></pre>
<pre><code>##     A0 w053.29.KC247625   .RAIOSATOA..HSONFT.E.-....E........</code></pre>
<pre><code>## Number of mutations to be represented is now 46</code></pre>
<pre><code>##   Checking for mutations in column 28: O193[YSN]</code></pre>
<pre><code>##     A0 w053.9.KC247633    DRGIOSIEIAG.HSONFT.E.TE....N..I....</code></pre>
<pre><code>## Number of mutations to be represented is now 44</code></pre>
<pre><code>##   Checking for mutations in column 31: T195[A]</code></pre>
<pre><code>## t=w078, n=33 viable clones</code></pre>
<pre><code>##   Checking for mutations in column  2: H499[S]</code></pre>
<pre><code>##   Checking for mutations in column  3: V359[S]</code></pre>
<pre><code>##   Checking for mutations in column  7: -214[TO]</code></pre>
<pre><code>##     A1 w078.9.KC247667    DRG.OSTAA.S.HSONFT.E....QK..S......</code></pre>
<pre><code>## Number of mutations to be represented is now 38</code></pre>
<pre><code>##     B0 w078.6.KC247664    DRGIOSOS.AS.HSONTN.OE-.............</code></pre>
<pre><code>## Number of mutations to be represented is now 36</code></pre>
<pre><code>##   Checking for mutations in column  8: -215[VK]</code></pre>
<pre><code>##   Checking for mutations in column 11: A216[VD]</code></pre>
<pre><code>##   Checking for mutations in column 14: N379[G]</code></pre>
<pre><code>##   Checking for mutations in column 17: -552[P]</code></pre>
<pre><code>##   Checking for mutations in column 18: -551[GE]</code></pre>
<pre><code>##   Checking for mutations in column 19: N544[SOD]</code></pre>
<pre><code>##     B0 w078.26.KC247645   DRGIOSTAAAS.HSON..S.O.E.NK..S......</code></pre>
<pre><code>## Number of mutations to be represented is now 33</code></pre>
<pre><code>##     A0 w078.33.KC247652   ..A.OSATOA..HSONT.O.N-.....N..I.T..</code></pre>
<pre><code>## Number of mutations to be represented is now 31</code></pre>
<pre><code>##     A1 w078.15.KC247637   DRGIOSATOA..HSON..D..TEL.KES.......</code></pre>
<pre><code>## Number of mutations to be represented is now 29</code></pre>
<pre><code>##   Checking for mutations in column 20: -550[TK]</code></pre>
<pre><code>##   Checking for mutations in column 21: K542[-]</code></pre>
<pre><code>##     A1 w078.29.KC247647   DRGIOSTAAAS.HSONTN..-..-L..NS.A....</code></pre>
<pre><code>## Number of mutations to be represented is now 27</code></pre>
<pre><code>##   Checking for mutations in column 24: I222[V]</code></pre>
<pre><code>##   Checking for mutations in column 25: H437[OGD]</code></pre>
<pre><code>##     B0 w078.17.KC247639   DRG.OSATOA..HSONFT.EE..LDK.D..IG...</code></pre>
<pre><code>## Number of mutations to be represented is now 25</code></pre>
<pre><code>##   Checking for mutations in column 27: G558[V]</code></pre>
<pre><code>##   Checking for mutations in column 28: O193[Y]</code></pre>
<pre><code>##   Checking for mutations in column 29: O218[KD]</code></pre>
<pre><code>##     A0 w078.30.KC247649   ..A.OSATOA..HSON....N-......D......</code></pre>
<pre><code>## Number of mutations to be represented is now 24</code></pre>
<pre><code>##   Checking for mutations in column 30: E727[ND]</code></pre>
<pre><code>##     A1 w078.36.KC247655   DRGIOSTAAAS.HSON..S.O-.-....SD.....</code></pre>
<pre><code>## Number of mutations to be represented is now 23</code></pre>
<pre><code>##   Checking for mutations in column 33:   M4[R]</code></pre>
<pre><code>##     A0 w078.27.KC247646   DRGIOSATOA..HSONTDD..TEL.KES....R..</code></pre>
<pre><code>## Number of mutations to be represented is now 22</code></pre>
<pre><code>## t=w100, n=26 viable clones</code></pre>
<pre><code>##   Checking for mutations in column  2: H499[S]</code></pre>
<pre><code>##     A1 w100.B10.KC247386      DSG.OSATOA..HSONTDD....LDKEN..I....</code></pre>
<pre><code>## Number of mutations to be represented is now 21</code></pre>
<pre><code>##   Checking for mutations in column  3: V359[S]</code></pre>
<pre><code>##     B0 w100.A11.KC247376      D.S.OSATOA..HSONTNTOE-..D.E.KD.....</code></pre>
<pre><code>## Number of mutations to be represented is now 19</code></pre>
<pre><code>##   Checking for mutations in column  8: -215[VK]</code></pre>
<pre><code>##     A1 w100.B2.KC247387   .RAIOSIK.AG.HSON....N...D.V........</code></pre>
<pre><code>## Number of mutations to be represented is now 17</code></pre>
<pre><code>##   Checking for mutations in column 11: A216[VD]</code></pre>
<pre><code>##     B0 w100.A13.KC247378      ..A.OSATOAV.HSONTNTOE-.......D.....</code></pre>
<pre><code>## Number of mutations to be represented is now 16</code></pre>
<pre><code>##   Checking for mutations in column 14: N379[G]</code></pre>
<pre><code>##   Checking for mutations in column 17: -552[P]</code></pre>
<pre><code>##   Checking for mutations in column 18: -551[GE]</code></pre>
<pre><code>##   Checking for mutations in column 20: -550[TK]</code></pre>
<pre><code>##   Checking for mutations in column 24: I222[V]</code></pre>
<pre><code>##   Checking for mutations in column 25: H437[OG]</code></pre>
<pre><code>##   Checking for mutations in column 28: O193[Y]</code></pre>
<pre><code>##   Checking for mutations in column 30: E727[N]</code></pre>
<pre><code>##   Checking for mutations in column 34: G707[SD]</code></pre>
<pre><code>##     A0 w100.T3.KC247401   DRGIOSATO..NHSONTDD.ETEL.KEN..I.RS.</code></pre>
<pre><code>## Number of mutations to be represented is now 15</code></pre>
<pre><code>##     A0 w100.B4.KC247389   DRGIOSATO...HSON..S.O...DKE.K....D.</code></pre>
<pre><code>## Number of mutations to be represented is now 14</code></pre>
<pre><code>## t=w136, n=28 viable clones</code></pre>
<pre><code>##   Checking for mutations in column  8: -215[V]</code></pre>
<pre><code>##     A1 w136.B2.KC247411   D.G.OSTVAA-.HGONIDOT--E.O.......RD.</code></pre>
<pre><code>## Number of mutations to be represented is now 10</code></pre>
<pre><code>##   Checking for mutations in column 11: A216[D]</code></pre>
<pre><code>##     A1 w136.B10.KC247404      D.GIOSATOADNHSONTD.E-TELDKES.DIY.S.</code></pre>
<pre><code>## Number of mutations to be represented is now 9</code></pre>
<pre><code>##   Checking for mutations in column 17: -552[P]</code></pre>
<pre><code>##   Checking for mutations in column 18: -551[GE]</code></pre>
<pre><code>##     B0 w136.B5.KC247429   ..A.OSATOAV.HSONTESK-.E.O..Y.DI....</code></pre>
<pre><code>## Number of mutations to be represented is now 6</code></pre>
<pre><code>##   Checking for mutations in column 24: I222[V]</code></pre>
<pre><code>##     B1 w136.B23.KC247414      D.A.OSIK..G.HSONTEST-..VD....N...D.</code></pre>
<pre><code>## Number of mutations to be represented is now 4</code></pre>
<pre><code>##   Checking for mutations in column 25: H437[G]</code></pre>
<pre><code>## t=w160, n=20 viable clones</code></pre>
<pre><code>##   Checking for mutations in column 17: -552[P]</code></pre>
<pre><code>##     A1 w160.T4.KC247483   D.A.OSTVA.S.HSONPD..-...G...DN.....</code></pre>
<pre><code>## Number of mutations to be represented is now 2</code></pre>
<pre><code>##   Checking for mutations in column 18: -551[G]</code></pre>
<pre><code>##     A1 w160.C1.KC247465   ..A.OSVTOAV.HSONTGST-...D..Y.D..TV.</code></pre>
<pre><code>## Number of mutations to be represented is now 1</code></pre>
<pre><code>##   Checking for mutations in column 35: R494[-]</code></pre>
<pre><code>##     C0 w160.T3.KC247482   D.SIOSATOA.NHSONTD.E-TELDKVND.IGRD-</code></pre>
<pre><code>## Number of mutations to be represented is now 0</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">    <span class="kw">summary</span>(eg.swarmset)</code></pre></div>
<pre><code>## Selected n=54 sequences:
## w000.TF 
## w004.31.KC247583, w004.54.KC247604 
## w007.8.KM284749, w007.21.KM284732, w007.25.KM284734, w007.34.KM284744 
## w008.20.KM284762 
## w009.19.KM284781 
## w010.7.KM284714 
## w020.15.KC247489, w020.11.KC247485, w020.24.KC247495, w020.25.KC247496 
## w022.6.KC247523, w022.5.KC247522, w022.9.KC247525, w022.22.KM284717 
## w030.20.KC247541, w030.17.KC247532, w030.21.KC247535, w030.36.KC247549, w030.26.KC247539, w030.13.KC247529, w030.32.KC247546 
## w053.15.KC247614, w053.29.KC247625, w053.22.KC247620, w053.8.KC247632, w053.31.KC247628, w053.9.KC247633 
## w078.6.KC247664, w078.36.KC247655, w078.9.KC247667, w078.26.KC247645, w078.29.KC247647, w078.30.KC247649, w078.33.KC247652, w078.17.KC247639, w078.15.KC247637, w078.27.KC247646 
## w100.T3.KC247401, w100.B10.KC247386, w100.B2.KC247387, w100.B4.KC247389, w100.A11.KC247376, w100.A13.KC247378 
## w136.B10.KC247404, w136.B5.KC247429, w136.B2.KC247411, w136.B23.KC247414 
## w160.C1.KC247465, w160.T3.KC247482, w160.T4.KC247483</code></pre>
<pre><code>## Covered 92 of 92 initial mutations.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">    <span class="kw">print</span>(eg.swarmset)</code></pre></div>
<pre><code>## w000.TF  NHVTNO---VADYNTK--N-KOKIHEGOOETDMGR 
## w004.31.KC247583 .....................-............. 
## w004.54.KC247604 K.................................. 
## w007.8.KM284749  KR................................. 
## w007.21.KM284732 ..................................Q 
## w007.25.KM284734 ............................N...... 
## w007.34.KM284744 ...I............................... 
## w008.20.KM284762 ..A................................ 
## w009.19.KM284781 ..G................................ 
## w010.7.KM284714  .N................................. 
## w020.15.KC247489 ....OS....T........................ 
## w020.11.KC247485 ..........TN....................... 
## w020.24.KC247495 .RA......AT........................ 
## w020.25.KC247496 .R....ATO.......................... 
## w022.6.KC247523  ..AIOSATO...H...................... 
## w022.5.KC247522  .RA.OSATO....S..................... 
## w022.9.KC247525  ..GIOS.................-........... 
## w022.22.KM284717 ...O............................... 
## w030.20.KC247541 ..AIOSATOA......TNTO............... 
## w030.17.KC247532 D.GOOSATO.......TD................. 
## w030.21.KC247535 .RA.OSATO...........E.............. 
## w030.36.KC247549 ....OSATO.....O.TD................. 
## w030.26.KC247539 .RG.OS....-....NTD................. 
## w030.13.KC247529 ..DPOS............................. 
## w030.32.KC247546 .RG.OSATO.......TD-................ 
## w053.15.KC247614 D.G.OSATOA..HSONFT.E.-.L........... 
## w053.29.KC247625 .RAIOSATOA..HSONFT.E.-....E........ 
## w053.22.KC247620 DRGIOSIEIAG.HSONFT.E.TE............ 
## w053.8.KC247632  DRGIOSATOA..HSONFT.E.T..Q.......... 
## w053.31.KC247628 DRGIOSAT....HSON....N-............. 
## w053.9.KC247633  DRGIOSIEIAG.HSONFT.E.TE....N..I.... 
## w078.6.KC247664  DRGIOSOS.AS.HSONTN.OE-............. 
## w078.36.KC247655 DRGIOSTAAAS.HSON..S.O-.-....SD..... 
## w078.9.KC247667  DRG.OSTAA.S.HSONFT.E....QK..S...... 
## w078.26.KC247645 DRGIOSTAAAS.HSON..S.O.E.NK..S...... 
## w078.29.KC247647 DRGIOSTAAAS.HSONTN..-..-L..NS.A.... 
## w078.30.KC247649 ..A.OSATOA..HSON....N-......D...... 
## w078.33.KC247652 ..A.OSATOA..HSONT.O.N-.....N..I.T.. 
## w078.17.KC247639 DRG.OSATOA..HSONFT.EE..LDK.D..IG... 
## w078.15.KC247637 DRGIOSATOA..HSON..D..TEL.KES....... 
## w078.27.KC247646 DRGIOSATOA..HSONTDD..TEL.KES....R.. 
## w100.T3.KC247401 DRGIOSATO..NHSONTDD.ETEL.KEN..I.RS. 
## w100.B10.KC247386    DSG.OSATOA..HSONTDD....LDKEN..I.... 
## w100.B2.KC247387 .RAIOSIK.AG.HSON....N...D.V........ 
## w100.B4.KC247389 DRGIOSATO...HSON..S.O...DKE.K....D. 
## w100.A11.KC247376    D.S.OSATOA..HSONTNTOE-..D.E.KD..... 
## w100.A13.KC247378    ..A.OSATOAV.HSONTNTOE-.......D..... 
## w136.B10.KC247404    D.GIOSATOADNHSONTD.E-TELDKES.DIY.S. 
## w136.B5.KC247429 ..A.OSATOAV.HSONTESK-.E.O..Y.DI.... 
## w136.B2.KC247411 D.G.OSTVAA-.HGONIDOT--E.O.......RD. 
## w136.B23.KC247414    D.A.OSIK..G.HSONTEST-..VD....N...D. 
## w160.C1.KC247465 ..A.OSVTOAV.HSONTGST-...D..Y.D..TV. 
## w160.T3.KC247482 D.SIOSATOA.NHSONTD.E-TELDKVND.IGRD- 
## w160.T4.KC247483 D.A.OSTVA.S.HSONPD..-...G...DN.....</code></pre>
<pre><code>## Number of mutations to be represented was initially 92.</code></pre>
<pre><code>## Number of mutations to be represented is now 0.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">    <span class="kw">plot</span>(eg.swarmset)</code></pre></div>
<pre><code>## [1] &quot;/tmp/RtmpFDrwBz/swarmset-n54271d78e1b023.png&quot;</code></pre>
<p>For some fun advanced plotting routines, try running <code>lassie::report.variant.frequencies(eg.swarmtools)</code> and <code>lassie::make.timepoint.logos(eg.swarmset)</code>.</p>
<hr />
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
